src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -386,7 +386,7 @@ public function get_items( $request ) {
 		$page        = (int) $query_args['paged'];
 		$total_posts = $posts_query->found_posts;
 
-		if ( $total_posts < 1 ) {
+		if ( $total_posts < 1 && $page > 1 ) {
```

Comment:
I've realized that this change will affect more than media, I also see you fixed some tests in post controller, TRAC ticket probably should be updated.

Reply:
Yeah that's what i discover. I guess it only happens whenever there's a completely empty collection of posts, which is more common with attachments than regular posts (given that they usually come with at least an example post). But it could happen for any post collection in theory!